src/options/dir_action.rs

```diff
@@ -54,79 +54,496 @@ impl RecurseOptions {
     /// flag’s value, and whether the `--tree` flag was passed, which was
     /// determined earlier. The maximum level should be a number, and this
     /// will fail with an `Err` if it isn’t.
-    pub fn deduce(matches: &MatchedFlags<'_>, tree: bool) -> Result<Self, OptionsError> {
-        if let Some(level) = matches.get(&flags::LEVEL)? {
-            let arg_str = level.to_string_lossy();
-            match arg_str.parse() {
-                Ok(l) => {
-                    Ok(Self { tree, max_depth: Some(l) })
-                }
-                Err(e) => {
-                    let source = NumberSource::Arg(&flags::LEVEL);
-                    Err(OptionsError::FailedParse(arg_str.to_string(), source, e))
-                }
-            }
-        }
-        else {
-            Ok(Self { tree, max_depth: None })
+    pub fn deduce(matches: &Opts, tree: bool) -> Result<Self, OptionsError> {
+        let max_depth = matches.level;
+        match max_depth {
+            Some(d) => {
+                Ok(Self {
+                    tree,
+                    max_depth: Some(d),
+                })
+            },
+            None => {
+                Ok(Self {
+                    tree,
+                    max_depth: None,
+                })
+            },
         }
     }
```

Comment:
Seems to be equivalent to:

```suggestion
    pub fn deduce(matches: &Opts, tree: bool) -> Result<Self, OptionsError> {
        Ok(
            Self,
            tree,
            max_depth: matches.level
        )
    }
```

so this function seems kinda useless now.